xpti/include/xpti_data_types.h

```diff
@@ -255,6 +269,10 @@ enum class trace_point_type_t : uint16_t {
   function_end = XPTI_TRACE_POINT_END(12),
   /// Use to notify that a new metadata entry is available for a given event
   metadata = XPTI_TRACE_POINT_BEGIN(13),
+  /// Used to trace function call begin and its arguments.
+  function_with_args_begin = XPTI_TRACE_POINT_BEGIN(14),
```

Comment:
Just FYI - In the future, you don't need to extend this enum. You can extend it as a "user_defined" trace point.